docs/setup-cluster/k8s/internal-task-gateway.rst

```diff
@@ -0,0 +1,128 @@
+.. _internal-task-gateway:
+
+#######################
+ Internal Task Gateway
+#######################
+
+`K8s Gateway APIs <https://gateway-api.sigs.k8s.io/>`_ allow us to expose otherwise internal
+Determined jobs running on remote K8s clusters to Determined master and proxies. This is useful for
+multi-resource manager setups.
+
+The overall setup includes installing and configuring a Gateway controller in the K8s cluster
+external to Determined and configuring the Determined master to use the Gateway controller. Please
+refer to the sections below to see the configuration changes and controller requirements.
+
+.. warning::
+
+   This feature exposes Determined tasks to the outside world. Please ensure that you have the
+   necessary security measures in place to both limit access to the exposed tasks, and secure the
+   communication between the external cluster and the main one. This could include setting up a
+   firewall, using a VPN, IP white-listing, K8s Network Policies, native cloud solutions, or other
+   security measures. Some Determined tasks including JupyterLab Notebooks and shells already have
+   secure transport built-in. Tensorboards do not use TLS currently so a tunneling solution should
+   be considered.
+
+Limitations:
+
+-  Proxying to multi-node trials is not supported. Currently this only includes experiments running
+      distributed training that want to manually expose proxies to the outside world.
+
+###################################
+ Controller Support - Requirements
+###################################
+
+High-level requirements from the controller implementations are:
+
+-  Supports Gateway APIs > v1
+
+-  Supports `TCPRoute
+   <https://gateway-api.sigs.k8s.io/concepts/api-overview/#tcproute-and-udproute>`_ and level 4
+   routing support
+
+Take a look at our current survey of the existing implementations here: :doc:`controller-reviews`
+
+In these docs, we'll be using `Contour from Project Contour <https://projectcontour.io/>`_.
+
+############################
+ Sample Setup - Development
+############################
+
+For internal testing and development, we provide a simple setup script that uses a dynamic
+provisioner provided by Project Contour.
+
+On a local dev machine, you can use Minikube and Contour as the controller. We provide a script to
+simplify the process. This can be found in `tools/k8s/launch-minikube-with-gateway.sh`.
+
+After you have a working K8s cluster and a Gateway controller running, configure the resource
+manager via master config and start the Determined cluster.
+
+###############
+ Configuration
+###############
+
+Below you'll find details on how to configure your cluster and Determined master to use the Internal
+Task Gateway.
```

Comment:
This section describes how to configure your cluster and Determined master to use the Internal Task Gateway.